providers/gce/gce_loadbalancer_internal.go

```diff
@@ -254,6 +256,55 @@ func (g *Cloud) ensureInternalLoadBalancer(clusterName, clusterID string, svc *v
 	return status, nil
 }
 
+func removeNodesInNonDefaultNetworks(nodes []*v1.Node, defaultSubnetName string) []*v1.Node {
+	var newList []*v1.Node
+	var skippedNodes []string
+	for _, node := range nodes {
+		subnetLabel, ok := node.Labels[labelGKESubnetworkName]
+		// nodes that have no label and no PodCIDR should be filtered out.
+		// This translates to: if node doesn't have the label but has PodCIDR then it is assumed to be in the default network.
+		// This check is a safeguard for situations when the cluster might be running older node controller that does not know multi-subnet or multi-subnet feature misbehaves.
+		if !ok && node.Spec.PodCIDR == "" {
+			skippedNodes = append(skippedNodes, node.Name)
+			continue
+		}
+		// For clusters that become multi-subnet the label on existing nodes from the default network can be present with an emtpy value.
+		if ok && subnetLabel != "" && subnetLabel != defaultSubnetName {
+			skippedNodes = append(skippedNodes, node.Name)
+			continue
+		}
+		newList = append(newList, node)
+	}
+	countOfSkippedNodes := len(skippedNodes)
+	if countOfSkippedNodes > 0 {
+		maxNodesToLog := 10
```

Comment:
You can probably inline this.
This is a const -- you don't need this as a variable.

Reply:
changed